Sources/Apollo/MultipartResponseParsingInterceptor.swift

```diff
@@ -0,0 +1,176 @@
+import Foundation
+#if !COCOAPODS
+import ApolloAPI
+#endif
+
+/// Parses multipart response data into chunks and forwards each on to the next interceptor.
+public struct MultipartResponseParsingInterceptor: ApolloInterceptor {
+
+  public enum MultipartResponseParsingError: Error, LocalizedError, Equatable {
+    case noResponseToParse
+    case cannotParseResponseData
+    case unsupportedContentType(type: String)
+    case cannotParseChunkData
+    case irrecoverableError(message: String?)
+    case cannotParsePayloadData
+
+    public var errorDescription: String? {
+      switch self {
+      case .noResponseToParse:
+        return "There is no response to parse. Check the order of your interceptors."
+      case .cannotParseResponseData:
+        return "The response data could not be parsed."
+      case let .unsupportedContentType(type):
+        return "Unsupported content type: application/json is required but got \(type)."
+      case .cannotParseChunkData:
+        return "The chunk data could not be parsed."
+      case let .irrecoverableError(message):
+        return "An irrecoverable error occured: \(message ?? "unknown")."
+      case .cannotParsePayloadData:
+        return "The payload data could not be parsed."
+      }
+    }
+  }
+
+  private enum ChunkedDataLine {
+    case heartbeat
+    case contentHeader(type: String)
+    case json(object: JSONObject)
+    case unknown
+  }
+
+  private let dataLineSeparator: String = "\r\n\r\n"
```

Comment:
If these are constants, they should be static, instead of initialized for each instance. 

nit: Also, you can make them `StaticString`, and use the `.description` property to turn them into `String` when using them for better performance.